include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -252,6 +253,8 @@ class IExecutionProvider {
    */
   virtual int GenerateMetaDefId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash) const;
 
+  virtual void RegisterAllocator(std::shared_ptr<AllocatorManager> allocator_manager);
```

Comment:
Can you add documentation for this new function? Why do we need it? Should users care to override it?